src/components/pages/TDBytes/ProductSuggestions.tsx

```diff
@@ -0,0 +1,167 @@
+import { deleteSuggestion, getSuggestions } from 'api/kiosk';
+import Icon from 'components/atoms/icons/icon';
+import ConfirmationBox from 'components/molecules/confirmationBox/ConfirmationBox';
+import Footer from 'components/molecules/footer/Footer';
+import Modal from 'components/molecules/modal/Modal';
+import { AuthenticateContext, Roles } from 'contexts/authProvider';
+import useModal from 'hooks/useModal';
+import useTitle from 'hooks/useTitle';
+import { useToast } from 'hooks/useToast';
+import { ProductSuggestion } from 'models/apiModels';
+import { useContext, useEffect, useState } from 'react';
+import { sortDate } from 'utils/sorting';
+import { transformDate } from 'utils/timeConverter';
+
+const SuggestionsPage = () => {
+  const [suggestions, setSuggestions] = useState<
+    ProductSuggestion[] | undefined
+  >(undefined);
+  const [selected, setSelected] = useState<ProductSuggestion | undefined>(
+    undefined
+  );
+  const [errMsg, setErrMsg] = useState<string | undefined>(undefined);
+
+  useTitle('TD Bytes - Forslag');
+  const { role } = useContext(AuthenticateContext);
+  const { addToast } = useToast();
+  const {
+    isOpen: isOpenDeleteModal,
+    onOpen: openDeleteModal,
+    onClose: closeDeleteModal,
+  } = useModal();
+
+  const deleteSelected = async () => {
+    try {
+      if (!selected) {
+        return;
+      }
+      await deleteSuggestion(selected.id);
+      setSuggestions(suggestions?.filter((s) => s.id !== selected.id));
+      addToast({
+        title: 'Suksess',
+        status: 'success',
+        description: 'Forslaget ble slettet',
+      });
+    } catch (error) {
+      if (error.statusCode === 404) {
+        addToast({
+          title: 'Feilmelding',
+          status: 'error',
+          description: 'Kunne ikke finne forslag i databasen',
+        });
+      } else {
+        addToast({
+          title: 'Feilmelding',
+          status: 'error',
+          description: 'En ukjent feil skjedde',
+        });
+      }
+    }
+    closeDeleteModal();
+  };
+
+  useEffect(() => {
+    const fetchSuggestions = async () => {
+      try {
+        const fetchedSuggestions = await getSuggestions();
+        const sortedSuggestions = fetchedSuggestions.sort((a, b) => {
+          /* Must correctly initialize date objects */
+          const aDate = new Date(a.timestamp);
+          const bDate = new Date(b.timestamp);
+          return sortDate(bDate, aDate); // Sort from newest to oldest
+        });
+        setSuggestions(sortedSuggestions);
+        setErrMsg(undefined);
+      } catch (error) {
+        if (error.statusCode === 404) {
+          setErrMsg('Fant ingen forslag...');
+        } else {
+          setErrMsg('Noe gikk galt');
+        }
+      }
+    };
+    /* Only allow authorized memebers to view suggestions */
+    if (role === Roles.admin || role === Roles.kiosk_admin) {
+      fetchSuggestions();
+    } else {
+      setErrMsg('Du har ikke de nødvendige rettighetene');
```

Comment:
Just a question. is it an error if I am on this page as a regular user? Shouldn't the fetch only happen if I have the right access and nothing else if I am a regular user?

Reply:
As this is the only kiosk_admin site so far, I think it's reasonable to do the auth check in the page instead of creating another route component to handle it